pkg/networking/middleware/auth_header.go

```diff
@@ -0,0 +1,50 @@
+package middleware
+
+import (
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/snyk/go-application-framework/internal/constants"
+	"github.com/snyk/go-application-framework/pkg/auth"
+	"github.com/snyk/go-application-framework/pkg/configuration"
+)
+
+type AuthHeaderMiddleware struct {
+	next          http.RoundTripper
+	authenticator auth.Authenticator
+	config        configuration.Configuration
+}
+
+func NewAuthHeaderMiddleware(
+	config configuration.Configuration,
+	authenticator auth.Authenticator,
+	roundTripper http.RoundTripper,
+) *AuthHeaderMiddleware {
+	return &AuthHeaderMiddleware{
+		next:          roundTripper,
+		config:        config,
+		authenticator: authenticator,
+	}
+}
+
+func (n *AuthHeaderMiddleware) RoundTrip(request *http.Request) (*http.Response, error) {
+	if request.URL != nil {
+		// determine configured api url
+		apiUrlString := n.config.GetString(configuration.API_URL)
+		apiUrl, err := url.Parse(apiUrlString)
+		if err != nil {
+			apiUrl, _ = url.Parse(constants.SNYK_DEFAULT_API_URL)
```

Comment:
Not sure, we should do this. if somebody configures an endpoint e.g. MT-EU, and they are accidentally re-routed to MT-US, that's not expected and regarding GDPR may lead to questions.

Reply:
there won't be re-routing based on this code but you are right, this is not the place to do the check, it already happens in the configuration.